extmod/modplatform.c

```diff
@@ -0,0 +1,285 @@
+ *
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2013-2021 Ibrahim Abdelkader <iabdalkader@openmv.io>
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+#include <stdint.h>
+#include <string.h>
+
+#include "py/runtime.h"
+#include "py/objtuple.h"
+#include "py/objstr.h"
+#include "genhdr/mpversion.h"
+
+#if MICROPY_PY_PLATFORM
+/// \module platform -  Access to underlying platform’s identifying data
+
+// TODO: Add more architectures, compilers and libraries.
+// See: https://sourceforge.net/p/predef/wiki/Home/
+
+#if defined(__ARM_ARCH)
+#define PLATFORM_ARCH   "arm"
+#elif defined(__x86_64__)
+#define PLATFORM_ARCH   "x86_64"
+#else
+#define PLATFORM_ARCH   ""
+#endif
+
+#if defined(__GNUC__)
+#define PLATFORM_COMPILER \
+    "GCC " \
+    MP_STRINGIFY(__GNUC__) "." \
+    MP_STRINGIFY(__GNUC_MINOR__) "." \
+    MP_STRINGIFY(__GNUC_PATCHLEVEL__)
+#elif defined(__ARMCC_VERSION)
+#define PLATFORM_COMPILER \
+    "ARMCC " \
+    MP_STRINGIFY((__ARMCC_VERSION / 1000000)) "." \
+    MP_STRINGIFY((__ARMCC_VERSION / 10000 % 100)) "." \
+    MP_STRINGIFY((__ARMCC_VERSION % 10000))
+#else
+#define PLATFORM_COMPILER       ""
+#endif
+
+#if defined(__GLIBC__)
+#define PLATFORM_LIBC_LIB       "glibc"
+#define PLATFORM_LIBC_VER \
+    MP_STRINGIFY(__GLIBC__) "." \
+    MP_STRINGIFY(__GLIBC_MINOR__)
+#elif defined(__NEWLIB__)
+#define PLATFORM_LIBC_LIB       "newlib"
+#define PLATFORM_LIBC_VER       _NEWLIB_VERSION
+#else
+#define PLATFORM_LIBC_LIB       ""
+#define PLATFORM_LIBC_VER       ""
+#endif
+
+#if defined(__linux)
+#define PLATFORM_SYSTEM     "Linux"
+#elif defined(__unix__)
+#define PLATFORM_SYSTEM     "Unix"
+#elif defined(__CYGWIN__)
+#define PLATFORM_SYSTEM     "Cygwin"
+#elif defined(__WIN32__)
+#define PLATFORM_SYSTEM     "Windows"
+#else
+#define PLATFORM_SYSTEM     "MicroPython"
+#endif
+
+#ifndef MICROPY_HAL_VERSION
+#define MICROPY_HAL_VERSION \
```

Comment:
what is this for, and why does it default to 1.2.0?

Reply:
It's just a place holder for HAL/SDK version so the module doesn't fail to build because it's not yet defined in any port... Should it default to "0.0.0" or fail to build if MICROPY_HAL_VERSION is not defined ?